source/src/in/joind/DateHelper.java

```diff
@@ -16,13 +17,14 @@
      */
     public static String parseAndFormat(String input, String format) {
         // Assume that the input format is ISO8601
-        SimpleDateFormat dfOutput = new SimpleDateFormat(format), dfInput = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+        SimpleDateFormat dfOutput = new SimpleDateFormat(format, Locale.US),
```

Comment:
You are expecting a date that agrees with the Date Format that it's used. If you ignore the Locale definition, the date returned from the smartphone might not contain the expected information.